cmake/Dependencies.cmake

```diff
@@ -47,23 +47,37 @@ if(USE_LIBUV)
         libuv_LIBRARY
         NAMES uv libuv
         HINTS ${libuv_ROOT} ENV libuv_ROOT
-        PATH_SUFFIXES lib/release lib/debug
+        PATH_SUFFIXES lib
         REQUIRED
         NO_DEFAULT_PATH)
-      if(NOT EXISTS ${libuv_LIBRARY})
-        message(FATAL_ERROR "Unable to find static libuv library in " ${libuv_ROOT})
+      if(EXISTS ${libuv_LIBRARY})
+        set(libuv_HEADER_SEARCH_PATH ${libuv_LIBRARY}/../../include)
+        set(libuv_DLL_PATH ${libuv_LIBRARY}/../../bin/uv.dll)
+      else()
+        find_library(
```

Comment:
curious, there is already a `find_library` before the `if` clause, why do we need to find it again here? Is it because the files sometimes live in `lib` and sometimes live in `lib/release` and `lib/debug?

cc @malfet, does this look OK to you?

Reply:
For libuv installed from conda, uv and uv_a libs are under lib folder, but if we build Libuv project and install on WIndows, uv and uv_a libs will under folder named lib\release or lib\debug, depends on the build config. Since when building gloo in PyTorch, need conda install libuv. But in Windows CI of Gloo, I follow the style of other CI, build the libuv from source code and install it. That is why I made it as this. If it is OK that Windows CI of Gloo works with installing libuv from conda source. That would be simple here. @malfet, how do you think?

Reply:
resolved